### DISCUSSION TITLE
How can I add Menu?

### DISCUSSION THREAD

**Discussion by reuse-ay** [maxGraph/maxGraph#399] [Q&A] [answered]
I'm trying to add Menu to my project by angular

I defined (copied) custom menu handler class by reference to [this](https://github.com/maxGraph/maxGraph/blob/main/packages/html/stories/MenuStyle.stories.js) from line 155 and add some import, but it's not going well. `menu` in factorymethod function don't have `addItem` and `addSeparator`. How can I resolve?

By the way, `MaxPopupMenu` has `addItem` and `addSeparator`. Is it relevant?

Minimum reproducible example:[here](https://stackblitz.com/edit/stackblitz-starters-utqs64?description=An%20angular-cli%20project%20based%20on%20@angular/animations,%20@angular/common,%20@angular/compiler,%20@angular/core,%20@angular/forms,%20@angular/platform-browser,%20@angular/platform-browser-dynamic,%20@angular/router,%20core-js,%20rxjs,%20tslib%20and%20zone.js&file=package.json,src%2Fmenu.ts&template=node&title=Angular%20Starter)

**Comment by tbouffard** [comment]
Hi, about the `factoryMethod`, this seems due to #308.

**Comment by reuse-ay** [comment, marked as answer]
Temporary solution:
`this.factoryMethod = function (menu_, cell, evt) {
      const menu=menu_ as unknown as MaxPopupMenu`

**Reply by tbouffard** [reply]
Does this workaround fix your problem?

**Reply by reuse-ay** [reply]
At the moment, yes. Menu is shown when clicked and alert in `addItem` also works in my project. But that's all I checked.
